src/btree/bt_compact.c

```diff
@@ -361,20 +361,17 @@ __wt_compact(WT_SESSION_IMPL *session)
           session, btree_compact_pages_rewritten_expected, stats_pages_rewritten_expected);
 
         /*
-         * Periodically check if we've timed out or eviction is stuck. Quit if eviction is stuck,
-         * we're making the problem worse.
+         * Periodically check if:
+         * - compaction has timed out,
+         * - eviction is stuck or,
+         * - compaction has been interrupted.
+         * Quit if any of the above is true.
          */
         if (first || ++i > 100) {
             if (!first)
                 bm->compact_progress(bm, session, &msg_count);
             WT_ERR(__wt_session_compact_check_timeout(session));
```

Comment:
Would it make a sense to move this check into `__wt_session_compact_check_interrupted`? I can see it either way—just was wondering what others think.

Reply:
FWIW, I have a (mild) preference for merging the two checks. My thinking is that if we want to add a new kind of check in the future we would have to update the code in two places right now. But if there is one generic function that checks for all of the reasons we might interrupt compaction, then there will just be one place to change the code.

Reply:
I do like the idea of merging them both! See the merge in f2409890526de1d9553a79bb0d433f413eff9bed.
Note that this impacted the LSM code as well, see f1f67426d290b83a218ba72933933adbf0dc824a.